src/main/java/net/mcreator/ui/minecraft/states/StateMap.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.minecraft.states;
+
+import com.google.gson.*;
+import com.google.gson.annotations.JsonAdapter;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This object holds the state map. It is a map of property (property name and its description) to value of the state.
+ */
+@JsonAdapter(StateMap.GSONAdapter.class) public class StateMap extends LinkedHashMap<IPropertyData<?>, Object> {
+
+	private static final Map<String, Class<? extends IPropertyData<?>>> typeMappings = new HashMap<>() {{

```

Comment:
I think I should eventually seal `IPropertyData` class as I only meant it to be implemented by `PropertyData` (for further extension by custom property types) and `BuiltInPropertyData` (for wrapping properties created from data list entries), and change this map type accordingly.

Reply:
What do you mean by seal it?

Reply:
Let's get the critical parts working again before we start nitpicking and delaying the PR again